### PR TITLE
Rest API User Groups

### PR DESCRIPTION
**Pull request by sachscode**
_Issue #, if available:_

_Description of changes:_
1. REST API : Description of these user types below - Added User Groups
2. REST API :  Added example "Create REST API and restrict specific routes to specific user groups"
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pages/cli/restapi/restapi.mdx [hunk 1]
```diff
@@ -135,6 +141,56 @@ REST
           |_ Logs (Amazon CloudWatch)
 ```
 
+## Create REST API and restrict specific routes to specific user groups
+If your app uses user-pool groups to manage different user types and would like to restrict access of specific routes to specifc user groups. we can accomplish this by the following flow:
```
Comment (siegerts) [inline src/pages/cli/restapi/restapi.mdx:145]:
```suggestion
If your app uses user-pool groups to manage different user types and would like to restrict access of specific routes to specifc user groups. We can accomplish this by the following flow: 
```

Comment (siegerts) [inline src/pages/cli/restapi/restapi.mdx:145]: Case convention should be standardized with above reference. User Pool Groups vs. user-pool

File: src/pages/cli/restapi/restapi.mdx [hunk 2]
```diff
@@ -135,6 +141,56 @@ REST
...
+
+>
+> The following example flow assumes the existence of two user pools AdminUsers, GuestUsers for a Book store.
+> The app would like to limit admin functionality like updating book records to the AdminUsers user-group, 
+> while borrowing and returning books would be limited to the GuestUsers user-group. 
+> - Path : /book/admin is restricted to AdminUsers and commands are handled by the bookAdminHandler lambda function
+> - Path : /book/guest is restricted to GuestUsers and commands are handled by the bookGuestHandler lambda function
+
+```bash
+amplify-dev add api
```
Comment (siegerts) [inline src/pages/cli/restapi/restapi.mdx:158]:
```suggestion
amplify add api
```

File: src/pages/cli/restapi/restapi.mdx [hunk 3]
```diff
@@ -135,6 +141,56 @@ REST
...
+$> ? Do you want to edit the local lambda function now? No
+Successfully added resource bookGuestHandler locally.
+$> ✔ Restrict API access? (Y/n) · yes
+$> ✔ Restrict access by: Individual Groups
+$> ✔ Select groups: GuestUsers
+$> ✔ What permissions do you want to grant to GuestUsers users?  create, read, update
+$> ✔ Do you want to add another path? (y/N) No
+✅ Successfully added resource mybookapi locally
+```
+At the end of this command you can verify the routes and their respective user group restrictions in cli-inputs.json file at the following path.
```
Comment (siegerts) [inline src/pages/cli/restapi/restapi.mdx:189]:
```suggestion
At the end of this command you can verify the routes and their respective user group restrictions in the `cli-inputs.json` file at the following path.
```

File: src/pages/cli/restapi/restapi.mdx [hunk 4]
```diff
@@ -135,6 +141,56 @@ REST
           |_ Logs (Amazon CloudWatch)
 ```
 
+## Create REST API and restrict specific routes to specific user groups
+If your app uses user-pool groups to manage different user types and would like to restrict access of specific routes to specifc user groups. we can accomplish this by the following flow: 
+- Create API route. 
+- Add API route handler function.
+- Restrict-access to the API route to the user-group. The user-group
```
Comment (siegerts) [inline src/pages/cli/restapi/restapi.mdx:148]:
```suggestion
- Restrict-access to the API route to the user-group.
```

File: src/pages/cli/restapi/restapi.mdx (hunk 1 above)
Comment (sachscode) [reply src/pages/cli/restapi/restapi.mdx]: Got it, will replace all user-pool references with User Pool.

**Review by siegerts — APPROVED** [review]

**Review by lazpavel — APPROVED** [review]
LGTM

**Pull request merged**